pkg/registry/apis/datasource/register.go

```diff
@@ -111,9 +108,14 @@ func NewDataSourceAPIBuilder(
 		return nil, err
 	}
 
+	client, ok := plugin.Client()
```

Comment:
^^ no real benefit yet 🤣  and this assumes all the same middleware is applied in this version vs getting filled in from the global (manager) plugins.Client